src/components/inputs/Button.tsx

```diff
@@ -1,23 +1,31 @@
 import React from 'react';
 import {Button as MuiButton, Icon, Box} from '@mui/material';
+import PropTypes from 'prop-types';
+import {DashComponentProps} from 'props';
 
 /**
  * Button component
  */
-const Button = (props) => {
-  const {
-    id,
-    text,
-    margin,
-    width,
-    nClicks,
-    startIcon,
-    endIcon,
-    setProps,
-    iconColor,
-    ...remainingProps
-  } = props;
-
+const Button = ({
+  id = 'button',
+  text,
+  margin = 2,
+  width = null,
+  nClicks = 0,
+  startIcon = null,
+  endIcon = null,
+  setProps,
+  iconColor,
+  variant = 'contained',
+  color = 'primary',
+  size = 'medium',
+  disabled = false,
+  disableRipple,
+  disableFocusRipple,
+  disableElevation,
+  href,
```

Comment:
We should remove the last props so they are unpacked into remainingProps and correctly passed into the MuiButton component, as now they are completely ignored by it.

```diff
- variant = 'contained',
- color = 'primary',
-  size = 'medium',
-  disabled = false,
-  disableRipple,
-  disableFocusRipple,
-  disableElevation,
-  href,
```

Reply:
well seen !